client/test/unit/components/bhLocationSelect.spec.js

```diff
@@ -0,0 +1,201 @@
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7E4', name : 'village 4', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430B6'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7E5', name : 'village 5', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430B7'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7E6', name : 'village 6', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430B7'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7E7', name : 'village 7', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430B8'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7E8', name : 'village 8', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430B8'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7E9', name : 'village 9', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430B9'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F1', name : 'village 10', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430B9'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F2', name : 'village 11', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C1'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F3', name : 'village 12', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C1'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F4', name : 'village 13', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C2'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F5', name : 'village 14', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C2'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F6', name : 'village 15', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C3'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F7', name : 'village 16', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C3'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F8', name : 'village 17', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C4'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7F9', name : 'village 18', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C4'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A1', name : 'village 19', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C5'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A2', name : 'village 20', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C5'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A3', name : 'village 21', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C6'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A4', name : 'village 22', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C6'},
+
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A5', name : 'village 23', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C7'},
+    {uuid : '0x03A329B203FE4F73B40F56A2870CC7A6', name : 'village 24', sector_uuid : '0x2C90F3C84EB146D39A5CDA180D1430C7'},
+  ]
+};
+
+describe('component Template and controller : bhLocationSelect', function () {
+  var componentController, element, scope, $compile;
+  var template = `<bh-location-select id="location-id" location-uuid="testData.villages[0].uuid"></bh-location-select>`;
```

Comment:
Instead of using `testData.villages[0].uuid`, a more meaningful way of writing this could be:
```js
const template = `
  <bh-location-select location-uuid="initialVillage">
  </bh-location-select>
`;

//  The uuid of test village 1
scope.initialVillage = '0x03A329B203FE4F73B40F56A2870CC7E1';
```
That way, we can easily change the village without rewriting the template, and we can dynamically swap the loaded village in each test by simply changing a variable on the scope.